blog/content/post/2024-12-13-ram-data-decay-research.md

```diff
@@ -0,0 +1,317 @@
+which memory is compared against expected pattern. If it doesn't match, whole
+page is removed from tested ranges. After that, the proper testing can be
+continued as originally planned.
+
+Another issue was that the memory map sometimes changed. In most cases it was
+consistent, but on every 10th or so run the application was loaded to a
+different address, most likely because for reasons unknown another allocation
+has changed. To work around this, we filtered out regions smaller than 16 MB,
+and aligned the remaining ones to 16 MB base and a size that is multiple of that
+value. This didn't fix the underlying problem, but it significantly reduced the
+likelihood of it causing problems.
+
+After few initial attempts we also resigned from precisely measuring DRAM
+temperatures. They were relatively constant and very close to ambient
+temperature on desktop, and on laptop they got hotter with time, even on idle.
+We think it may be caused by GPU located in close proximity, but we haven't
+tested similar laptop without graphics card yet.
+
+As for ambient temperature, first iterations were run in our office, with
+temperature momentarily getting as hot as 24 &#8451;. It was impossible to get
+any reasonable data in these conditions for laptop, so we've moved testing to
+air-conditioned lab with temperatures oscillating between 19 to 20 &#8451;.
+
+## Tested platforms
+
+We've decided to test two platforms, different both because one is a laptop, and
+another is a PC, and because one uses DDR5, and the other - DDR4. Because of
+that, comparison between both doesn't make sense, but that was not our goal.
+
+First tested platform was a NovaCustom laptop, V540TND, with W-NM56S508G DDR5
+SODIMM. As a PC, we used MSI PRO Z790-P WIFI DDR4 with Kingston KF432C16BB/4
+DIMM. As you can see, we've chosen small memory sizes for testing. While the
+results could be slightly different with more memory modules installed due to
+worse air flow, at this point we wanted to save time, since we weren't sure if
+we would get any useful results in the first place.
+
+If there is enough interest, we may extend our tests to cover more
+configurations in the future.
+
+## Results
+
+Below are the results, divided by the platform, which is also a division between
+DDR generations. At this point we don't have enough data to draw a conclusion as
+to whether such drastically different results are caused by platform type, DIMM
+variant or memory generation.
+
+For all tests, air humidity was constant 33%. Temperatures were changing
+slightly, they are listed below next to the results.
+
+Results show number of changed bits, divided between transitions from 1 to 0 and
+from 0 to 1, as well as an average of the two.
+
+### Laptop - DDR5 SODIMM
+
+On this platform, we've struggled to get any preserved bits at all. We ended up
+disconnecting the battery and unplugging and, as quickly as humanely possible,
```

Comment:
```suggestion
disconnecting the battery and unplugging and, as quickly as humanly possible,
```

Reply:
https://github.com/3mdeb/news-and-ideas/pull/621/commits/478ab8793dac42e71f71d1b2f30aeb7b92d053c6